NGPIris/cli/functions.py

```diff
@@ -12,26 +12,57 @@
 from pathlib import Path
 
 from NGPIris import log, TIMESTAMP
-from NGPIris.hcp import HCPManager
 from NGPIris.preproc import preproc
 
 ##############################################
 
+
 @click.command()
 @click.argument("query")
-@click.option("-m", "--mode",help="Restrict search to a file type", type=click.Choice(['all','file', 'dir'], case_sensitive=False),default='all')
+@click.option("-i","--index",help="NGPi index name")
+@click.option("-o","--output",help="Specify output file to write to",default="")
+@click.option("-v" "--verbose",is_flag=True,default=False)
+@click.option("-m", "--mode",help="Search mode", type=click.Choice(['ngpi','ngpr'], case_sensitive=False),default='ngpr')
 @click.pass_obj
-def search(ctx, query, mode):
-    """List all file hits for a given query"""
-    if query != "":
+def search(ctx, index, query, verbose,mode):
+    """List all file hits for a given query by directly calling HCP"""
+  
+    if not query != "":
+        log.info('A query or file needs to be specified if you are using the "search" option')
+        sys.exit(-1)
+
+    if mode == "ngpr":
         found_objs = ctx['hcpm'].search_objects(query,mode=mode)
         if len(found_objs) > 0:
             for obj in found_objs:
                 log.info(obj.key)
         else:
             log.info(f'No results found for: {query}')
-    else:
-        log.info('A query or file needs to be specified if you are using the "search" option')
+
+    elif mode == "ngpi":
+        hcim = ctx['hcim']
+
+        hcim.create_template(index, query)
+        token = hcim.generate_token()
+
+        if verbose:
+            resp = hcim.query(token)
```

Comment:
`.query()` should really take the generated query json path from previously as input.
Now the connection relies solely on the file path written and read match.